### DISCUSSION TITLE
How to make fal run when dbt models are executed conditionally

### DISCUSSION THREAD

**Discussion by ADP-BA** [fal-ai/dbt-fal#175] [Q&A]
Hello, 

I've got a dbt project which has several models. Not all of the models are related and I do not want all of them to run on each "dbt run" execution. For that purpose, I've added the "enabled" property in the models passing them a variable so that each of them executes when needed. 

Something like this:

```
{{ config(
		schema='dwh',
		materialized= 'incremental', 
		unique_key= 'id',
                enabled=var('sectionX'),
		...)
}}
```

I then added the meta section in the schema.yml in order to run the python script when the model has ran. Right now I've got a very simple py file that only creates a file, just to test fal. 

After all that I've run dbt as always:
`dbt run --vars '{ "sectionX": "true", ... }'`

I've checked the 'run_results.json' and I can see that the model to which I've attached the python script has been successfully executed.

```
{
      "status": "success",
      "timing": [
        {
          "name": "compile",
          "started_at": "2022-03-06T08:45:51.497657Z",
          "completed_at": "2022-03-06T08:45:51.509625Z"
        },
        {
          "name": "execute",
          "started_at": "2022-03-06T08:45:51.509625Z",
          "completed_at": "2022-03-06T08:45:51.633294Z"
        }
      ],
      "thread_id": "Thread-1",
      "execution_time": 0.13763189315795898,
      "adapter_response": { "_message": "OK", "rows_affected": 0 },
      "message": "OK",
      "failures": null,
      "unique_id": "model.dwh.TestTable"
    }
```

And then run fal (even using the --all flag). But the python script is not running and I always get the same message: 

```
08:58:20  Unable to do partial parsing because config vars, config profile, or config target have changed
08:58:21  Found 0 models, 0 tests, 0 snapshots, 0 analyses, 190 macros, 0 operations, 9 seed files, 0 sources, 0 exposures, 0 metrics
09:58:21 | Starting fal run for following models and scripts: 


```

But, as can be seen above, the list of models and scripts is empty and so the python script is not being executed. 

Am I doing something wrong? 
Or what I am trying to achieve is not possible?

Thanks!

**Comment by chamini2** [comment]
Hey, @ADP-BA thanks for reaching out!

I think the way you are executing the commands is fine: `fal run` looks at your run_results.json and then runs the appropriate scripts based on that.

What may be happenning is that the `schema.yml` for assigning the Python scripts as post-hook of models has an issue. 

Can you paste a model you have in your schema.yml along with it's `meta` property to see how you are specifying the scripts to run?

BTW, here is documentation about how that should be: https://docs.fal.ai/Docs/fal-cli/model-scripts
